### PR TITLE
Add Property expression for integer multiplication.

### PR DESCRIPTION
**Pull request by mikeurbach**
### Contributor Checklist

- [x] Did you add Scaladoc to every public function/method?
- [x] Did you add at least one test demonstrating the PR?
- [x] Did you delete any extraneous printlns/debugging code?
- [x] Did you specify the type of improvement?
- [x] Did you add appropriate documentation in `docs/src`?
- [x] Did you request a desired merge strategy?
- [x] Did you add text to be included in the Release Notes for this change?

#### Type of Improvement

- Feature (or new API)

#### Desired Merge Strategy

- Squash: The PR will be squashed and merged (choose this if you have no preference).

#### Release Notes

This adds an API for integer Property multiplication.

### Reviewer Checklist (only modified by reviewer)
- [ ] Did you add the appropriate labels? (Select the most appropriate one based on the "Type of Improvement")
- [ ] Did you mark the proper milestone (Bug fix: `3.6.x`, `5.x`, or `6.x` depending on impact, API modification or big change: `7.0`)?
- [ ] Did you review?
- [ ] Did you check whether all relevant Contributor checkboxes have been checked?
- [ ] Did you do one of the following when ready to merge:
  - [ ] Squash: You/ the contributor `Enable auto-merge (squash)`, clean up the commit message, and label with `Please Merge`.
  - [ ] Merge: Ensure that contributor has cleaned up their commit history, then merge with `Create a merge commit`.

### CODE REVIEW

File: core/src/main/scala/chisel3/properties/Property.scala [hunk 1]
```diff
@@ -323,6 +323,9 @@ sealed trait Property[T] extends Element { self =>
 
   final def +(that: Property[T])(implicit ev: PropertyArithmeticOps[Property[T]], sourceInfo: SourceInfo): Property[T] =
```
Comment (mwachs5) [inline core/src/main/scala/chisel3/properties/Property.scala:324]:
```suggestion
  /** Perform addition as defined by FIRRTL spec section 25.1.1. (as of 4.0.0) */
  final def +(that: Property[T])(implicit ev: PropertyArithmeticOps[Property[T]], sourceInfo: SourceInfo): Property[T] =
```

or maybe there is a more future-proof way of referring to that

File: core/src/main/scala/chisel3/properties/Property.scala [hunk 2]
```diff
@@ -323,6 +323,9 @@ sealed trait Property[T] extends Element { self =>
...
     ev.add(this, that)
+
+  final def *(that: Property[T])(implicit ev: PropertyArithmeticOps[Property[T]], sourceInfo: SourceInfo): Property[T] =
+    ev.mul(this, that)
```
Comment (mwachs5) [inline core/src/main/scala/chisel3/properties/Property.scala:328]: can you copy the great info from `properties.md` as scaladoc here?

File: docs/src/explanations/properties.md [hunk 3]
```diff
@@ -143,6 +143,7 @@ class IntegerArithmeticExample extends RawModule {
 The following table lists the possible arithmetic operators that are supported
 on integral `Property` typed values.
 
-| Operation | Description                                                |
-| --------- | -----------                                                |
-| `+`       | Perform addition as defined by FIRRTL spec section 25.1.1. |
+| Operation | Description                                                      |
+| --------- | -----------                                                      |
+| `+`       | Perform addition as defined by FIRRTL spec section 25.1.1.       |
```
Comment (mwachs5) [inline docs/src/explanations/properties.md:148]: nit: it seems likely that these section numbers would move around; are we planning to keep updating this? Maybe we can refer to the section by name instead which seems likely to be more stable?

File: core/src/main/scala/chisel3/properties/Property.scala (hunk 2 above)
Comment (mikeurbach) [reply core/src/main/scala/chisel3/properties/Property.scala:328]: Ah, yes I'll add scaladoc to the public methods, thanks for pointing it out.

File: docs/src/explanations/properties.md (hunk 3 above)
Comment (mikeurbach) [reply docs/src/explanations/properties.md:148]: hmm yeah, they probably will drift. i think referencing by name sounds good.

**Review by mwachs5 — APPROVED** [review]

**Pull request merged**